chain/jsonrpc-primitives/src/errors.rs

```diff
@@ -59,18 +80,84 @@ impl RpcError {
     }
     /// Create an invalid request error.
     pub fn invalid_request() -> Self {
```

Comment:
It is only used in a single place, and it indicates that the user submitted not a JSON RPC request (`{"jsonrpc": "2.0", "id": "qwe", "method": "asd", "params": {}}`), but something else, so I suggest we just use `ParseError` with a message that JSON RPC Request format was expected.